cmd/certificates/auth.go

```diff
@@ -0,0 +1,273 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/certificates"
+	"storj.io/storj/pkg/cfgstruct"
+	"storj.io/storj/pkg/utils"
+)
+
+var (
+	authCmd = &cobra.Command{
+		Use:   "auth",
+		Short: "CSR authorization management",
+	}
+
+	authCreateCmd = &cobra.Command{
+		Use:   "create <auth_increment_count> [<email>, ...]",
+		Short: "Create authorizations from a list of emails",
+		Args:  cobra.MinimumNArgs(1),
+		RunE:  cmdCreateAuth,
+	}
+
+	authInfoCmd = &cobra.Command{
+		Use:   "info [<email>, ...]",
+		Short: "Get authorization(s) info from CSR authorization DB",
+		RunE:  cmdInfoAuth,
+	}
+
+	authExportCmd = &cobra.Command{
+		Use:   "export [<email>, ...]",
+		Short: "Export authorization(s) from CSR authorization DB to a CSV file (or stdout)",
+		RunE:  cmdExportAuth,
+	}
+
+	authCreateCfg struct {
+		certificates.CertServerConfig
+		batchCfg
+	}
+
+	authInfoCfg struct {
+		ShowTokens bool `help:"if true, token strings will be printed" default:"false"`
+		certificates.CertServerConfig
+		batchCfg
+	}
+
+	authExportCfg struct {
+		All bool   `help:"export all authorizations" default:"false"`
+		Out string `help:"output file path; if \"-\", will use STDOUT" default:"$CONFDIR/authorizations.csv"`
+		certificates.CertServerConfig
+		batchCfg
+	}
+)
+
+func init() {
+	rootCmd.AddCommand(authCmd)
+	authCmd.AddCommand(authCreateCmd)
+	cfgstruct.Bind(authCreateCmd.Flags(), &authCreateCfg, cfgstruct.ConfDir(defaultConfDir))
+	authCmd.AddCommand(authInfoCmd)
+	cfgstruct.Bind(authInfoCmd.Flags(), &authInfoCfg, cfgstruct.ConfDir(defaultConfDir))
+	authCmd.AddCommand(authExportCmd)
+	cfgstruct.Bind(authExportCmd.Flags(), &authExportCfg, cfgstruct.ConfDir(defaultConfDir))
+}
+
+func cmdCreateAuth(cmd *cobra.Command, args []string) error {
+	count, err := strconv.Atoi(args[0])
```

Comment:
is there any concern args will ever be empty?

Reply:
He specified a minimum above in L34. That should prevent it from being empty